test/Concurrency/sendable_conformance_checking.swift

```diff
@@ -0,0 +1,147 @@
+// RUN: %target-typecheck-verify-swift
+// REQUIRES: concurrency
+
+@available(SwiftStdlib 5.1, *)
+class NotSendable { // expected-note 8{{class 'NotSendable' does not conform to the 'Sendable' protocol}}
+}
+
+@available(SwiftStdlib 5.1, *)
+@available(*, unavailable)
+extension NotSendable: Sendable { }
+
+@available(SwiftStdlib 5.1, *)
+protocol IsolatedWithNotSendableRequirements: Actor {
+  func f() -> NotSendable
+  var prop: NotSendable { get }
+}
+
+// Okay, everything is isolated the same way
+@available(SwiftStdlib 5.1, *)
+actor A1: IsolatedWithNotSendableRequirements {
+  func f() -> NotSendable { NotSendable() }
+  var prop: NotSendable { NotSendable() }
+}
+
+// Okay, sendable checking occurs when calling through the protocol
+// and also inside the bodies.
+@available(SwiftStdlib 5.1, *)
+actor A2: IsolatedWithNotSendableRequirements {
+  nonisolated func f() -> NotSendable { NotSendable() }
+  nonisolated var prop: NotSendable { NotSendable() }
+}
+
+@available(SwiftStdlib 5.1, *)
+protocol AsyncProtocolWithNotSendable {
+  func f() async -> NotSendable
+  var prop: NotSendable { get async }
+}
+
+// Sendable checking required because calls through protocol cross into the
+// actor's domain.
+@available(SwiftStdlib 5.1, *)
+actor A3: AsyncProtocolWithNotSendable {
+  func f() async -> NotSendable { NotSendable() } // expected-warning{{cannot call function returning non-sendable type 'NotSendable' across actors}}
```

Comment:
the error message is super confusing here...? There is no call here 🤔

Reply:
Yeah, we aren't customizing these diagnostics well as a general thing.